src/Control/Email/Email.php

```diff
@@ -31,738 +18,641 @@ class Email extends ViewableData
-            }
-        }
+        return $this;
+    }
 
-        if ($sendAllfrom = $this->config()->send_all_emails_from) {
-            if ($from) {
-                $subject .= " [from $from]";
-            }
-            $from = $sendAllfrom;
-        }
+    public function setFailedRecipients($recipients)
+    {
+        $this->failedRecipients = $recipients;
 
-        Requirements::restore();
+        return $this;
+    }
 
-        return self::mailer()->sendHTML(
-            $to,
-            $from,
-            $subject,
-            $this->body,
-            $this->attachments,
-            $headers,
-            $this->plaintext_body
-        );
+    public function getFailedRecipients()
+    {
+        return $this->failedRecipients;
     }
 
     /**
-     * Validates the email address to get as close to RFC 822 compliant as possible.
+     * Used by {@link SSViewer} templates to detect if we're rendering an email template rather than a page template
      *
-     * @param string $email
      * @return bool
+     */
+    public function IsEmail()
+    {
+        return true;
+    }
+
+    /**
+     * Send the message to the recipients
      *
-     * @copyright Cal Henderson <cal@iamcal.com>
-     *  This code is licensed under a Creative Commons Attribution-ShareAlike 2.5 License
-     *  http://creativecommons.org/licenses/by-sa/2.5/
+     * @return bool true if successful or array of failed recipients
      */
-    public static function is_valid_address($email)
+    public function send()
     {
-        $qtext = '[^\\x0d\\x22\\x5c\\x80-\\xff]';
-        $dtext = '[^\\x0d\\x5b-\\x5d\\x80-\\xff]';
-        $atom = '[^\\x00-\\x20\\x22\\x28\\x29\\x2c\\x2e\\x3a-\\x3c'.
-            '\\x3e\\x40\\x5b-\\x5d\\x7f-\\xff]+';
-        $quoted_pair = '\\x5c[\\x00-\\x7f]';
-        $domain_literal = "\\x5b($dtext|$quoted_pair)*\\x5d";
-        $quoted_string = "\\x22($qtext|$quoted_pair)*\\x22";
-        $domain_ref = $atom;
-        $sub_domain = "($domain_ref|$domain_literal)";
-        $word = "($atom|$quoted_string)";
-        $domain = "$sub_domain(\\x2e$sub_domain)*";
-        $local_part = "$word(\\x2e$word)*";
-        $addr_spec = "$local_part\\x40$domain";
+        $this->getSwiftMessage()->setContentType('text/html');
+        if (!$this->getBody()) {
+            $this->render();
+        }
+        //create plain text part
+        $this->getSwiftMessage()->addPart(
+            Convert::xml2raw($this->getSwiftMessage()->getBody()),
+            'text/plain',
+            'utf-8'
+        );
+        return Injector::inst()->get(Mailer::class)->send($this);
+    }
 
-        return preg_match("!^$addr_spec$!", $email) === 1;
+    /**
+     * @return array|bool
+     */
+    public function sendPlain()
+    {
+        $this->getSwiftMessage()->setContentType('text/plain');
+        if (!$this->getBody()) {
+            $this->render();
+            $this->setBody(Convert::xml2raw($this->getBody()));
```

Comment:
Alternatively `getPlainTextSwiftMessage()` could be a second getter that transforms the email prior to passing it to the mailers. This, however, would mean that the mailer would need to accept a swift message object rather than our email. I don't have a problem with that, but it's another tweak to the interface.